### PR TITLE
GP-14 - Popup panel templates

### PR DESCRIPTION
**Pull request by branflake2267**
Two presenter widget types for popup, one for the built in gwt panel, and second for a custom popup panel. 

I'll aim to parameterize the custom popuppanel/presenterwidget a bit more during that phase, but this provides the basic aim.

### CODE REVIEW

File: src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__View.java.vm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import com.google.web.bindery.event.shared.EventBus;
+import com.gwtplatform.mvp.client.ViewWithUiHandlers;
+import com.gwtplatform.mvp.client.ViewImpl;
+
+public class ${name}View extends $extends implements ${name}Presenter.MyView {
+    public interface Binder extends UiBinder<Widget, ${name}View> {
+    }
+
+    @UiField
+    PopupPanel popup;
```
Comment (branflake2267) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__View.java.vm:23]: will paramertize this so any type of messaging dialog or independent widget can fit here.

Comment (branflake2267) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__View.java.vm:23]: Or maybe not. Probably better make its own home.

Comment (branflake2267) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__View.java.vm:23]: nevermind, the presenter widget it self could become what I'm thinking. :)

File: src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__Presenter.java.vm [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    public interface MyView extends View ${extendsView} {
+        void center();
+
+        void hide();
+
+        void showRelativeTo(Widget target);
+    }
+
+    @Inject
+    public ${name}Presenter(EventBus eventBus, MyView view) {
```
Comment (christiangoudreau) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__Presenter.java.vm]: Remove public (Guice best practice)

File: src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__View.java.vm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Inject
+    public ${name}View(Binder binder, EventBus eventBus) {
```
Comment (christiangoudreau) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/custom/__name__View.java.vm]: remove public

File: src/main/resources/com/arcbees/plugin/template/presenter/popup/gwt/__name__PresenterWidget.java.vm [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import com.google.web.bindery.event.shared.EventBus;
+import com.gwtplatform.mvp.client.PopupView;
+import com.gwtplatform.mvp.client.PresenterWidget;
+
+public class ${name}PresenterWidget extends PresenterWidget<${name}PresenterWidget.MyView> ${implements} {
+    public interface MyView extends PopupView ${extendsView} {
+    }
+
+    @Inject
+    public ${name}PresenterWidget(final EventBus eventBus, final MyView view) {
```
Comment (christiangoudreau) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/gwt/__name__PresenterWidget.java.vm]: remove public

File: src/main/resources/com/arcbees/plugin/template/presenter/popup/gwt/__name__View.java.vm [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import com.google.web.bindery.event.shared.EventBus;
+import com.gwtplatform.mvp.client.PopupViewImpl;
+import com.gwtplatform.mvp.client.PopupViewWithUiHandlers;
+
+public class ${name}View extends $extends implements ${name}PresenterWidget.MyView {
+    public interface Binder extends UiBinder<PopupPanel, ${name}View> {
+    }
+
+    @Inject
+    public ${name}View(Binder uiBinder, EventBus eventBus) {
```
Comment (christiangoudreau) [inline src/main/resources/com/arcbees/plugin/template/presenter/popup/gwt/__name__View.java.vm]: remove public

File: src/main/resources/com/arcbees/plugin/template/presenter/widget/__name__Presenter.java.vm [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+import com.gwtplatform.mvp.client.HasUiHandlers;
+import com.gwtplatform.mvp.client.PresenterWidget;
+import com.gwtplatform.mvp.client.View;
+
+public class ${name}Presenter extends PresenterWidget<${name}Presenter.MyView> ${implements} {
+    public interface MyView extends View ${extendsView} {
+    }
+
+    @Inject
+    public ${name}Presenter(EventBus eventBus, MyView view) {
```
Comment (christiangoudreau) [inline src/main/resources/com/arcbees/plugin/template/presenter/widget/__name__Presenter.java.vm]: remove public

File: src/main/resources/com/arcbees/plugin/template/presenter/widget/__name__View.java.vm [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+public class ${name}View extends ${extends} implements ${name}Presenter.MyView {
+    public interface Binder extends UiBinder<HTMLPanel, ${name}View> {
+    }
+
+    @UiField
+    HTMLPanel panel;
+
+    @Inject
+    public ${name}View(Binder binder) {
```
Comment (christiangoudreau) [inline src/main/resources/com/arcbees/plugin/template/presenter/widget/__name__View.java.vm]: remove public

**Comment by branflake2267** [comment]
Do the interfaces need public still?

**Comment by branflake2267** [comment]
all cr fixes in this last commit :)

**Comment by christiangoudreau** [comment]
Interface doesn't need to be public as well.

**Pull request merged**